### DISCUSSION TITLE
Generated types for MYSQL integers

### DISCUSSION THREAD

**Discussion by mooxiu** [aarondl/sqlboiler#1412] [General]
Due to the nature of golang's design of `int`, `int32` and `int64`, sometimes it is very confusing to generate a type of `int`. In my opinion, if we can decide it is a 32bit number, I would argue generating an `int32` type rather than `int` type is preferable.

[Here in MYSQL 8.4, they have integer types of 1, 2, 3, 4, 8 bytes](https://dev.mysql.com/doc/refman/8.4/en/integer-types.html). While the generating rule in `drivers/sqlboiler-mysql/driver/mysql.go` about the 4 byte integer `INT` looks like this (Master branch as of now Aug 28th 2024):

```go
case "int", "integer":
			if unsigned {
				c.Type = "null.Uint"
			} else {
				c.Type = "null.Int"
			}
```

When mysql stores a 4 byte integer number, the generated type is `Int` or `Uint`. Considering nowadays most of us are using 64-bit systems, then the generated type will be a 8 byte integer, some static checking tools like `gosec` would complain if we're trying to cast this to `int32`. Considering people use protobuf a lot and it [requires to assign`int32` and `int64` explicitly](https://protobuf.dev/programming-guides/proto3/), it is not rare that people would do such an casting operation.

In short, I would suggest to change this to generate `null.Uint32` and `null.Int32` in this part.

**Comment by stephenafamo** [comment]
You're correct, however this would be a breaking change so cannot be done until `v5`

However, if you can send in a PR it would be appreciated so that it is not forgotten when v5 starts being worked on.

**Reply by mooxiu** [reply]
thanks for replying, I will try to make a PR for that
